From kuninori.morimoto.gx@gmail.com Tue Jun 12 21:51:37 2012
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 12 Jun 2012 21:51:35 -0700 (PDT)
Subject: [PATCH 2/2] LTSI: bugfix: sh73a0: add lost CLK_ENABLE_ON_INIT for DIV6_ZB1
To: Greg KH <gregkh@linuxfoundation.org>, ltsi-dev@lists.linuxfoundation.org
Cc: Simon <horms@verge.net.au>, Kuninori Morimoto <kuninori.morimoto.gx@gmail.com>, (KMC)tuji <kunihiko@kmckk.co.jp>
Message-ID: <87y5nr7r1o.wl%kuninori.morimoto.gx@renesas.com>


On upstream kernel, on one branch,
CLK_ENABLE_ON_INIT was added to DIV6_ZB1 clock by
9bcc0a5d0de137b3a154dc951c5ff70dce815879
(ARM: mach-shmobile: SH73A0 external Ethernet fix)

On the other branch, below commit modified dev6_clks[] controlling method.
d4775356bb39eaa305844cc6cc4c267236535956
(sh: clkfwk: clock-sh73a0: all div6_clks use SH_CLK_DIV6_EXT())

These 2 commit had conflict, and were solved when merge window
timing on upstream kernel.
But unfortunately, this solution seems had be lost on backporting.
This patch fixes it up for LTSI.

Reported-by: Kunihiko Tsuji <kunihiko@kmckk.co.jp>
Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 arch/arm/mach-shmobile/clock-sh73a0.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-shmobile/clock-sh73a0.c b/arch/arm/mach-shmobile/clock-sh73a0.c
index 012b284..fcecbdc 100644
--- a/arch/arm/mach-shmobile/clock-sh73a0.c
+++ b/arch/arm/mach-shmobile/clock-sh73a0.c
@@ -328,7 +328,7 @@ static struct clk div6_clks[DIV6_NR] = {
 			vck_parent, ARRAY_SIZE(vck_parent), 12, 3),
 	[DIV6_VCK3] = SH_CLK_DIV6_EXT(VCLKCR3, 0,
 			vck_parent, ARRAY_SIZE(vck_parent), 12, 3),
-	[DIV6_ZB1] = SH_CLK_DIV6_EXT(ZBCKCR, 0,
+	[DIV6_ZB1] = SH_CLK_DIV6_EXT(ZBCKCR, CLK_ENABLE_ON_INIT,
 			pll_parent, ARRAY_SIZE(pll_parent), 7, 1),
 	[DIV6_FLCTL] = SH_CLK_DIV6_EXT(FLCKCR, 0,
 			pll_parent, ARRAY_SIZE(pll_parent), 7, 1),
-- 
1.7.5.4

